### PR TITLE
[Bug fix] FHIR invalid results display name

### PR DESCRIPTION
**Pull request by zdeveloper**
# BACKEND PULL REQUEST

## Related Issue

- The bulk uploader is not setting the correct result display name for the result SNOMED

## Changes Proposed

- fix the bug
- Add ndjson tests

### CODE REVIEW

File: backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java [hunk 1]
```diff
@@ -229,14 +228,8 @@ private InputStream getJsonStream(String jsonFile) {
...
   }
 
-  private String readJsonStream(InputStream is) throws IOException {
-    BufferedReader reader = new BufferedReader(new InputStreamReader(is));
-    StringBuilder stringBuilder = new StringBuilder();
-    String line;
-    while ((line = reader.readLine()) != null) {
-      stringBuilder.append(line);
-    }
-    return stringBuilder.toString();
```
Comment (zdeveloper) [inline backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java:239]: This method was eating all new lines in the input stream.

**Review by mpbrown — APPROVED** [review]
LGTM!

File: backend/src/test/resources/testResultUpload/test-results-upload-valid-different-results.csv [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+4567,Doe,Janey,,123 Main St,,Birmingham,AL,35226,Jefferson,205-999-2800,1/20/1962,F,White,Not Hispanic or Latino,,,567,ID NOW,94534-5,Invalid Result,12/20/2021 14:00,12/21/2021 14:00,12/22/2021 14:00,12/23/2021 14:00,12/24/2021 14:00,Nasal Swab,1013012657,Smith MD,John,,400 Main Street,,Birmingham,AL,35228,205-888-2000,01D1058442,My Testing Lab,300 North Street,,Birmingham,AL,35228,205-888-2000,N,N,N,,N,,N,N,My Urgent Care,400 Main Street,Suite 100,Birmingham,AL,35228,205-888-2000,Test Comment,,F
+4567,Doe,Janey,,123 Main St,,Birmingham,AL,35226,Jefferson,205-999-2800,1/20/1962,F,White,Not Hispanic or Latino,,,678,ID NOW,94534-5,260373001,12/20/2021 14:00,12/21/2021 14:00,12/22/2021 14:00,12/23/2021 14:00,12/24/2021 14:00,Nasal Swab,1013012657,Smith MD,John,,400 Main Street,,Birmingham,AL,35228,205-888-2000,01D1058442,My Testing Lab,300 North Street,,Birmingham,AL,35228,205-888-2000,N,N,N,,N,,N,N,My Urgent Care,400 Main Street,Suite 100,Birmingham,AL,35228,205-888-2000,Test Comment,,F
```
Comment (zdeveloper) [inline backend/src/test/resources/testResultUpload/test-results-upload-valid-different-results.csv:7]: This file contains all the accepted values for test results 
- Positive
- Negative
- Detected
- Not Detected
- Invalid Result
- snomed value (260373001)

File: backend/src/test/resources/testResultUpload/test-results-upload-valid-as-fhir.ndjson [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
...[{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0078","code":"A","display":"Abnormal"}]}],"method":{"extension":[{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/testkit-name-id","valueCoding":{"code":"BOOMX2"}},{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/equipment-uid","valueCoding":{"code":"543212134"}}],"coding":[{"display":"ID NOW"}]},"specimen":{"reference":"Specimen/5db534ea-5e97-4861-ba18-d74acc46db15"},"device":{"reference":"Device/5db534ea-5e97-4861-ba18-d74acc46db15"}}},{"fullUrl":"Observation/a65bbde7-854d-3151-b46b-731badba996b","resource":{"resourceType":"Observation","id":"a65bbde7-854d-3151-b46b-731badba996b","identifier":[{"use":"official","type":{"coding":[{"system":"http://loinc.org","code":"81959-9","display":"Public health laboratory ask at order entry panel"}]}}],"status":"final","code":{"coding":[{"system":"http://loinc.org","code":"95419-8","display":"Has symptoms related to condition of interest"}],"text":"Has symptoms related to condition of interest"},"subject":{"reference":"Patient/4567"},"valueCodeableConcept":{"coding":[{"system":"http://terminology.hl7.org/ValueSet/v2-0136","code":"N","display":"No"}]}}}]}
```
Comment (zdeveloper) [inline backend/src/test/resources/testResultUpload/test-results-upload-valid-as-fhir.ndjson:6]: This file contains all the accepted values for test results in ndjson FHIR format

Positive
Negative
Detected
Not Detected
Invalid Result
snomed value (260373001) has an added `detected` as display value in the FHIR

File: backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java (hunk 1 above)
Comment (mpbrown) [reply backend/src/test/java/gov/cdc/usds/simplereport/utils/BulkUploadResultsToFhirTest.java:239]: Good catch for the multi row test files!

**Review by nathancrtr — APPROVED** [review]
Nice, thanks for getting this addressed so quickly!

**Review by DanielSass — APPROVED** [review]

**Pull request merged**